packages/backend/src/graphql/workHour/mutations/workHourUpdateMutationField.test.ts

```diff
@@ -327,4 +338,56 @@ describe('workHourUpdateMutationField', () => {
     expect(response.data).toBeNull()
     expect(response.errors).toEqual([new GraphQLError('Not authorized')])
   })
+
+  it('should update project member work hours when the updating user has admin role', async () => {
+    await prisma.workHour.create({
+      data: {
+        date: new Date('2022-01-01'),
+        duration: 30,
+        userId: '3',
+        taskId: 'T1',
+      },
+    })
+
+    const testServer = getTestServer({ userId: '1' })
+    const response = await testServer.executeOperation({
+      query: workHourUpdateMutation,
+      variables: {
+        data: {
+          date: '2022-01-01',
+          duration: 60,
+          taskId: 'T1',
+        },
+        date: '2022-01-01',
+        taskId: 'T1',
+        projectMemberUserId: '3',
+      },
+    })
+
+    expect(response.data).toEqual({
+      workHourUpdate: {
+        date: '2022-01-01',
+        duration: 60,
+        task: {
+          id: 'T1',
+          title: 'Task 1',
+        },
+        user: {
+          id: '3',
+          name: 'Test User with project membership and member role',
+        },
+      },
+    })
+    expect(response.errors).toBeUndefined()
+
+    const oldWorkHour = await prisma.workHour.findUnique({
+      where: { date_userId_taskId: { date: new Date('2022-01-02'), taskId: 'T1', userId: '3' } },
+    })
+    expect(oldWorkHour).toBeNull() // old work hour of project member is set to new date and should therefore not exist anymore
```

Comment:
Why do we check for `2022-01-02`? Since the date wasn't modified by the mutation

Reply:
It's not checking if the date has changed. It just makes sure that the work hour doesn't exist on that date anymore.